src/test/java/com/thomashan/coup/turn/state/WaitingChallengeMainActionStateTest.java

```diff
@@ -0,0 +1,17 @@
+package com.thomashan.coup.turn.state;
+
+import com.thomashan.coup.action.ChallengeAction;
+import com.thomashan.coup.action.ChallengeActionType;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class WaitingChallengeMainActionStateTest extends WaitingChallengeMainActionStateTestCase {
+    @Test
+    public void testPerformAction_GivenMainActionIsNotChallengeable_ThrowsException() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unit tests should not contain more than 1 assert(s).](https://app.codacy.com/app/thomashan/coup-java/pullRequest?prid=4561385)